### PR TITLE
Add write barriers to fixed anon constructors

### PR DESCRIPTION
**Pull request by RobDangerous**
And here is yet another blind fix based on your comment to the previous pull request.

PS: Just now also added a fix for the previous pull request.

### CODE REVIEW

**Comment by hughsando** [comment]
This is looking good, but you do not need write-barriers on the names of 'setFixed' or the AnonStructs because these are known to be generated from const strings allocated outside the GC system (hence no entries in the __Mark code too).  It should still work, but with a tiny performance overhead.

**Comment by RobDangerous** [comment]
Okidoki, removed the stringy write barriers.

File: include/hx/Anon.h [hunk 1]
```diff
@@ -81,10 +81,7 @@ class HXCPP_EXTERN_CLASS_ATTRIBUTES Anon_obj : public hx::Object
       fixed->key = inName;
       fixed->value = inValue;
       if (inValue.type == ::cpp::Variant::typeObject) {
-        HX_OBJ_WB_GET(this, hx::PointerOf(inValue.valObject));
-      }
-      else if (inValue.type == ::cpp::Variant::typeString) {
```
Comment (hughsando) [inline include/hx/Anon.h]: Actually, this one should stay.  The value could be a variable, while the name is known to be constant.

Comment (RobDangerous) [reply include/hx/Anon.h]: Fixed.

**Comment by hughsando** [comment]
Brilliant.  Thanks for the changes - no good deed goes unpunished.

**Comment by RobDangerous** [comment]
Thanks for all the help in getting this working!

**Pull request merged**